src/lib/components/inventory/list_item_modal.ts

```diff
@@ -0,0 +1,474 @@
+                                          ? 'Enter listing price in USD'
+                                          : 'Enter starting price in USD'}"
+                                  />
+                                  <input
+                                      type="range"
+                                      min="1"
+                                      max="200"
+                                      .value="${this.pricePercentage}"
+                                      @input="${this.handlePercentageChange}"
+                                      class="percentage-slider"
+                                  />
+                                  <div>Percentage of recommended price: ${this.pricePercentage.toFixed(1)}%</div>
+
+                                  ${this.listingType === 'auction'
+                                      ? html`
+                                            <div class="auction-settings">
+                                                <label>Auction Duration</label>
+                                                <select
+                                                    class="duration-select"
+                                                    .value="${this.auctionDuration}"
+                                                    @change="${(e: Event) =>
+                                                        (this.auctionDuration = Number(
+                                                            (e.target as HTMLSelectElement).value
+                                                        ) as 1 | 3 | 5 | 7 | 14)}"
+                                                >
+                                                    <option value="1">1 Day</option>
+                                                    <option value="3">3 Days</option>
+                                                    <option value="5">5 Days</option>
+                                                    <option value="7">7 Days</option>
+                                                    <option value="14">14 Days</option>
+                                                </select>
+                                            </div>
+                                        `
+                                      : ''}
+                              </div>
+
+                              ${this.error ? html`<div class="error-message">${this.error}</div>` : ''}
+
+                              <button
+                                  class="submit-button"
+                                  ?disabled="${this.isLoading || !this.customPrice}"
+                                  @click="${this.handleSubmit}"
+                              >
+                                  ${this.isLoading
+                                      ? 'Listing...'
+                                      : this.listingType === 'buy_now'
+                                      ? 'List for Sale'
```

Comment:
We'll probably need to add a confirmation popup of some sort (similar to the CSFloat UX) since users will accidentally click on this and it results in many support tickets.